syntax = "proto3";

package imageprocessing;

service ImageService {
    rpc CompareImages(CompareRequest) returns (CompareResponse);
}

message CompareRequest {
    bytes color_image = 1;
    repeated bytes bw_images = 2;
}

message CompareResponse {
    int32 matching_index = 1;
}